test/cluster/conftest.py

```diff
@@ -158,8 +165,40 @@ def cluster_con(hosts: list[IPAddress | EndPoint], port: int, use_ssl: bool, aut
                    )
 
 
-@pytest.fixture(scope="session")
-async def manager_internal(event_loop, request):
+@pytest.fixture(scope="module")
+async def manager_api_sock_path(request: pytest.FixtureRequest, testpy_test: Test) -> AsyncGenerator[str]:
+    if manager_api := request.config.getoption("--manager-api"):
+        yield manager_api
+    else:
+        test_uname = testpy_test.uname
+        clusters = testpy_test.suite.clusters
+        base_dir = str(testpy_test.suite.log_dir)
+        sock_path = f"{tempfile.mkdtemp(prefix='manager-', dir='/tmp')}/api"
+
+        start_event = Event()
+        stop_event = Event()
+
+        async def run_manager() -> None:
+            mgr = ScyllaClusterManager(test_uname=test_uname, clusters=clusters, base_dir=base_dir, sock_path=sock_path)
+            await mgr.start()
+            start_event.set()
+            try:
+                await asyncio.get_running_loop().run_in_executor(None, stop_event.wait)
+            finally:
+                await mgr.stop()
+
+        manager_process = Process(target=lambda: asyncio.run(run_manager()))
+        manager_process.start()
+        start_event.wait()
+
+        yield sock_path
+
+        stop_event.set()
+        manager_process.join()
+
+
+@pytest.fixture(scope="module")
```

Comment:
This patch changes "manager_internal" from session scope to module scope. Is it intentional? If it is, please explain why in the commit message.

Reply:
Yes, it's intentional. On this stage I'm trying to have a behavior as close to test.py as possible. test.py runs test file-by-file, so effectively scopes `session`, `package`, and `module` are the same.